File: planning-idsdp-pps-frontend/src/app/main/modules/rdpp-rtapp-management/features/dpp-tapp-list/project-details/services/effect-impact.service.ts
import {HttpClient} from '@angular/common/http';
import {Injectable} from '@angular/core';
import {Observable} from 'rxjs';
import {EffectImpactUrlConstant} from '../constants/effect-impact.url.constants';
import {EffectImpact} from '../models/effect-impact.model';
import {map} from 'rxjs/operators';

@Injectable({
    providedIn: 'root'
})
export class EffectImpactService {


    constructor(private _http: HttpClient) {
    }


    saveEffectImpact(effectImpact: EffectImpact): Observable<any> {
        return this._http.post(EffectImpactUrlConstant.CREATE_EFFECT_IMPACT, effectImpact);
    }

    saveEffectImpactAttachment(id: any, envClearenceAttachment: File): Observable<any> {
        const formData = new FormData();
        formData.append('envClearenceAttachment', envClearenceAttachment);
        formData.append('effectImpactId', id.toString());
        return this._http.post(EffectImpactUrlConstant.CREATE_EFFECT_IMPACT_ATTACHMENT, formData);
    }

    getEffectImpact(projectId) {
        return this._http.get(EffectImpactUrlConstant.GET_EFFECT_IMPACT + '/' + projectId).pipe(map((res: any) => {
            return res;
        }));
        ;
    }

    updateEffectImpact(projectId, model) {
        return this._http.put(EffectImpactUrlConstant.UPDATE_EFFECT_IMPACT + '/' + projectId, model).pipe(map((res: any) => {
            return res;
        }));
    }
}

File: planning-idsdp-pps-rdpp-rtapp-backend/src/main/java/com/ibcs/idsdp/rdpprtapp/services/EffectImpactService.java
package com.ibcs.idsdp.rdpprtapp.services;

import com.ibcs.idsdp.rdpprtapp.model.domain.EffectImpact;
import com.ibcs.idsdp.rdpprtapp.web.dto.request.EffectImpactRequest;

public interface EffectImpactService {

    public EffectImpact saveEffectImpact(EffectImpactRequest effectImpactRequest);

    EffectImpact getEffectImpact(Long effectImpactId);

    Boolean updateEffectImpact(EffectImpact effectImpact);

     EffectImpact getEffectImpact(String projectId);

     EffectImpact updateEffectImpact(EffectImpactRequest effectImpactRequest, String id);
}
